pkg/cfghandler/querytimeoutupdate.go

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// # This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cfghandler
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/siglens/siglens/pkg/config"
+	log "github.com/sirupsen/logrus"
+	"github.com/valyala/fasthttp"
+)
+
+type QueryTimeoutConfig struct {
+	TimeoutSecs int `json:"timeoutSecs"`
+}
+
+func GetQueryTimeout(ctx *fasthttp.RequestCtx) {
+	// Read the value from the runmod config file, but if that doesn't exist,
+	// read the value from the config file.
+	var timeoutSecs int
+	runModConfig, err := config.ReadRunModConfig(config.RunModFilePath)
+	if err != nil {
+		timeoutSecs = config.GetQueryTimeoutSecs()
+	} else {
+		timeoutSecs = runModConfig.QueryTimeoutSecs
+		if timeoutSecs == 0 {
+			timeoutSecs = config.GetQueryTimeoutSecs()
+		}
+	}
+
+	ctx.SetStatusCode(fasthttp.StatusOK)
+	ctx.SetContentType("application/json")
+	_, err = ctx.WriteString(`{"timeoutSecs":` + fmt.Sprintf("%v", timeoutSecs) + `}`)
+	if err != nil {
+		log.Errorf("GetQueryTimeout: Error writing response: %v", err)
+		ctx.SetStatusCode(fasthttp.StatusInternalServerError)
+		return
+	}
+}
+
+func SaveQueryTimeoutToRunMod(filepath string, timeoutSecs int) error {
+	existingConfig, err := config.ReadRunModConfig(filepath)
+	if err != nil {
+		log.Errorf("SaveQueryTimeoutToRunMod: Could not read existing config from %s: %v", filepath, err)
+	}
+
+	var configData struct {
+		PQSEnabled       bool `json:"pqsEnabled"`
+		QueryTimeoutSecs int  `json:"queryTimeoutSecs"`
+	}
```

Comment:
nit: you've duplicated this struct also in pqsupdate.go. If we add another configurable thing to the runmod config, we'll have to add a field to this struct and the one in pqsupdate.go and copy the struct in the new `func Save...ToRunMod()`. It would be better to define this type outside the functions; then you can define it just once, and if we add a new configurable item we'll have to only update the single type definition (and add the functions to read/write the new field)

Reply:
Actually, there's already the `RunModConfig` type in config/common.go. I think you can use that here